src/node.cc

```diff
@@ -254,6 +254,11 @@ std::string config_userland_loader;  // NOLINT(runtime/string)
 // is used.
 bool config_pending_deprecation = false;
 
+// Set by EnvSetter when a warning is emitted for non-string assignment to
+// process.env, to prevent the same warning from being emitted multiple times
+// per process.
+bool env_nonstring_deprecation_warned = false;
```

Comment:
Rather than add more global state can this be a flag on `Environment`?

Reply:
Sure.